lldb/test/API/functionalities/watchpoint/modify-watchpoints/main.c

```diff
@@ -0,0 +1,17 @@
+#include <stdint.h>
+int main() {
+  int value = 5;
+
+  value = 5; // break here
+  value = 5;
+  value = 5;
+  value = 5;
+  value = 5;
+  value = 5;
+  value = 10;
+  value = 10;
```

Comment:
Can we check that it stops when the value changes from 10 to something else to ensure we detect all watchpoint changes and we can ensure that m_new_value_sp and newest_valueobj_sp are doing their jobs correctly? We need to make sure we stop on all changes, here we are testing on change

Reply:
We could change it back to 5 to ensure that the watchpoint detects when we change from 5 to 10 (and stops) and then back to 5 (it should stop again because it changed from 10 to 5).